tests/cli/utils.py

```diff
@@ -128,3 +129,9 @@ def assert_no_tracebacks(click_result):
     except ValueError as ve:
         # sometimes stderr is not captured separately
         pass
+
+
+def escape_ansi(line: str) -> str:
```

Comment:
If this get called repeatedly on the string we could speed up the tests with a `functool.lru_cache` here.
```suggestion
@functools.lru_cache(maxsize=64)
def escape_ansi(line: str) -> str:
```